From 5f0f7744d2d240755937936b407da163ab6a4128 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Mon, 16 Apr 2018 19:40:13 +0800
Subject: [PATCH] arm64: rk3308_linux_defconfig: disable SPARSEMEM_VMEMMAP

Save about 15MB memory for 64MB device, 8MB for 512MB.

For 64MB device,
before:
Memory: 37984K/63488K available (5182K kernel code, 366K rwdata,
 1740K rodata, 256K init, 521K bss, 25504K reserved, 0K cma-reserved)

after:
Memory: 53348K/63488K available (5182K kernel code, 366K rwdata,
 1740K rodata, 256K init, 521K bss, 10140K reserved, 0K cma-reserved)

For 512MB device,
before:
Memory: 495992K/522240K available (5182K kernel code, 366K rwdata,
 1740K rodata, 256K init, 521K bss, 26248K reserved, 0K cma-reserved)

after:
Memory: 504188K/522240K available (5182K kernel code, 366K rwdata,
 1740K rodata, 256K init, 521K bss, 18052K reserved, 0K cma-reserved)

Change-Id: Ica3431aff50a52e3d66d9e203f2313fcfdca068b
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 arch/arm64/configs/rk3308_linux_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/configs/rk3308_linux_defconfig b/arch/arm64/configs/rk3308_linux_defconfig
index 295f2d96a8cf..a1a2b5f5e219 100644
--- a/arch/arm64/configs/rk3308_linux_defconfig
+++ b/arch/arm64/configs/rk3308_linux_defconfig
@@ -33,6 +33,7 @@ CONFIG_SCHED_MC=y
 CONFIG_NR_CPUS=4
 CONFIG_PREEMPT=y
 CONFIG_HZ_1000=y
+# CONFIG_SPARSEMEM_VMEMMAP is not set
 # CONFIG_COMPACTION is not set
 CONFIG_DEFAULT_MMAP_MIN_ADDR=32768
 # CONFIG_UNMAP_KERNEL_AT_EL0 is not set
-- 
2.35.3

